app/Actions/Update/Apply.php

```diff
@@ -45,7 +45,7 @@ private function check_prod_env_allow_migration(array &$output): bool
 		if (Config::get('app.env') === 'production') {
 			// @codeCoverageIgnoreStart
 			// we cannot code cov this part. APP_ENV is dev in testing mode.
-			if (Configs::getValueAsBool('force_migration_in_production')) {
+			if (Configs::getValueAsBool('force_migration_in_production', false)) {
```

Comment:
As @kamil4 said we should really decide wether we want to bother with explicit defaults everywhere.

However, we should come to a clear decision and be consistent about it. If we want to use explicit default values everywhere, then we should remove the implicit default value from the definition of `getValueAsBool` and enforce this style. At the moment my IDE is rather unhappy, that we provide the same value for the 2nd parameter which is defined as default value anyway. If we decide to not bother with explicit default values, then we could as well remove the 2nd parameter entirely. Alternatively, we could also throw an exception, if the configuration parameter does not exist. I would ask for a vote:

 1. Enforce that explicit default values must be provided everywhere as done here. This also means to remove the implicit default value from the definition of the method `getValueAsBool` (same holds for the other methods as well)
 2. Don't care at all and return a pre-defined default value silently, if the the configuration parameter does not exist in the DB. This would mean to remove the 2nd parameter entirely even from the definition of the method `getValueAsBool` and friends.
 3. Don't provide a default value, remove the 2nd parameter entirely and throw an exception, if the config parameter is unknown. The latter should actually only happen during development or if a user manipulated the database intentionally.

I am in favor or option 3.

Rationale: Option 1 is cumbersome. Moreover it bears the risk of inconsistencies. I.e. in we could accidentally use a different default value somwhere than somewhere else. This may lead to unexpected program behavior. Option 2 sounds to risky for me. If a parameter does not exist and we return a hard-coded default value for it, that default value might be bad choice. Let's take `getValueAsBool` as an example. I don't believe that either returning `false` or `true` as the global default matches all configuration parameters well.

Reply:
3. Same rationale.

Reply:
1. is the closest to what we've been doing until now, sans enforcement. And since it's the one that actually made me question why we are doing things this way, I guess I'm for option 3 as well.